### ISSUE TITLE
Accessibility Testing for MHV on VA, Medical Records, Download your medical records reports

### ISSUE THREAD

**Issue by BobbyBaileyRB** [department-of-veterans-affairs/va.gov-team#98932]
### Product information

- [x] Team name, product name, and feature name have been added to the title of this issue.
- [x] Team label, product label and feature label (if applicable) have been added to this issue.

### Who completed the use of color and color contrast test?

BobbyBaileyRB

### Use of color and color contrast checks

- [x] All text of 20px or smaller has a 4.5:1 contrast ratio to its background (or better)
- [x] All text of 20px or larger has a 3:1 contrast ratio to its background (or better)
- [x] Non-text elements have a 3:1  contrast ratio to their background and to neighboring elements (or better)
- [x] Color is not the only way to distinguish links from other text (eg. links are underlined)
- [x] Any charts, maps, infographics, and tables convey all information without only relying on color
- [x] Content does not refer to color, especially when providing user instructions (eg. "Click the blue button")

### Who completed the axe scans?

BobbyBaileyRB

### axe checks

- [x] Each page has been scanned using axe (results shared below)
- [x] axe is integrated into your end-to-end testing

### axe DevTools scan results

![Image](https://github.com/user-attachments/assets/64483391-629d-4e09-adb7-67061e6e861d)

![Image](https://github.com/user-attachments/assets/990f3def-5ee4-4450-add7-178f76e9fd95)

![Image](https://github.com/user-attachments/assets/917e9fe9-b493-49bc-9310-dc1601073d42)

![Image](https://github.com/user-attachments/assets/579090e4-7083-4137-a646-e8c602c8938e)

### Who completed the content zoom and reflow test?

BobbyBaileyRB

### Content zoom and reflow checks

- [x] All page elements are readable and usable at 200% zoom
- [x] All page elements are readable and usable at 300% zoom
- [x] All page elements are readable and usable at 400% zoom

### Who completed the keyboard navigation test?

BobbyBaileyRB

### Keyboard navigation checks

- [x] Each link, button, form input, checkbox, radio button, select menu, and custom element can receive keyboard focus
- [x] Each link, button, form input, checkbox, radio button, select menu, and custom element responds to expected keys
- [x] All elements under focus have a visible focus indicator
- [x] The order of [Tab] stops made sense and was appropriate for completing tasks

### How did keyboard testing go?

In the VADS for checkboxes. People cannot use the Enter key to select a checkbox, is this the desired behavior?

### Do you have any other results to share?

**Other Reports You Can Download**
- Issue: Incorrect heading structure in the "Download your medical records" accordion section. The "Other reports you can download" heading is an H2, but the accordions beneath it are also H2s.
- Recommendation: Update the HTML (not the visual design) so the accordions are H3s, making them subordinate to the H2 heading.

**Title Page for Download Your Medical Records**
- Issue: The title of the page is currently <title>MHV Medical Records | Veterans Affairs</title>. This is not unique and does not match the H1 of the page.
- Recommendation: Ensure the title tag is unique and aligns with the H1 heading of the page for consistency and better accessibility.
Select Records and Download Report

**Potential VA Forms Library Issue:**
- Issue 1: Heading Structure: Skips from H1 to H3, which breaks the logical hierarchy.
- Issue 2: When navigating between pages, the focus state does not land on any elements or headers. For instance, moving from page 1 ("1 of 3 Select date range") to page 2 ("2 of 3 Select types of records to include") does not correctly shift focus to help screen reader navigation and understand where they are on the page.

**Select Date Range**
- Issue: Selecting a custom date range shows no data for the days, and the select component is partially cut off.

**Select Types of Records to Include**
- Issue: The checkboxes do not have a visible title or meaningful legend field could be based on how the forms are built.

**Select File Type**
- Issue 1: No error message is shown if neither of the two radio buttons is selected.
- Issue 2: There is a known issue in VADS where the incorrect number of radio buttons is announced.

**All Flows for Select Records and Download Report**
- Issue: When making a selection and navigating back and forth between steps, the selection is not retained.
- Issue: Focus management when an error is display, the focus stays on the continue button and doesn't go to the first error in question.

**Comment by bryan-riley-va** [comment]
Created JIRA MHV-65064
